networking_cisco/plugins/cisco/db/l3/l3_router_appliance_db.py

```diff
@@ -179,9 +179,11 @@ def _cisco_router_result_filter_hook(self, query, filters):
     def do_create_router(self, context, router, router_type_id, auto_schedule,
                          share_host, hosting_device_id=None, role=None,
                          inflated_slot_need=0):
+        r = router['router']
+        gw_info = r.pop(EXTERNAL_GW_INFO, None)
```

Comment:
This addresses a problem encountered with the query hook. Creating a router with the gateway set was failing. The failure was happening in the create_port_precommit call of our APIC ML2 mechanism driver (creating a router with the gateway set causes the gateway port to be created as well). The reason that was failing was because it was trying to get the router that owns the gateway. The get was failing because there is a hook added by the Cisco Router Service Plugin, which filters queries for routers with the HostingDevice extension. Unfortunately, the hosting device extension isn't added until after the router itself is created, so the query was failing, causing the port create to fail, which failed the whole router create to fail as well.